File: apps/mobile/lib/domain/usecases/school/get_classes_by_school_usecase.dart
import '../../../core/error/failures.dart';
import '../../../core/usecases/usecase.dart';
import '../../../core/utils/either.dart';
import '../../entities/class/class_selector_entity.dart';
import '../../repositories/school/class_repository.dart';

class GetClassesBySchoolUseCase
    implements
        UseCase<
          Either<Failure, List<ClassSelectorEntity>>,
          GetClassesBySchoolParams
        > {
  final ClassRepository repository;

  GetClassesBySchoolUseCase(this.repository);

  @override
  Future<Either<Failure, List<ClassSelectorEntity>>> call(
    GetClassesBySchoolParams params,
  ) async {
    return await repository.getClassesBySchool(schoolId: params.schoolId);
  }
}

class GetClassesBySchoolParams {
  final String schoolId;

  GetClassesBySchoolParams({required this.schoolId});
}

File: services/user_service/src/domain/usecases/class/GetClassesBySchoolUseCase.ts
import { IClassRepository } from "../../repositories/IClassRepository";

export class GetClassesBySchoolUseCase {
  constructor(private classRepository: IClassRepository) {}

  async execute(params: { schoolId: string; grade?: string }): Promise<any[]> {
    return await this.classRepository.getClassesBySchool(params);
  }
}
